src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const current = this[i];
+
+      if (current !== null && current !== undefined) {
+        result += current;
+      }
+
+      if (i < this.length - 1) {
+        result += separator !== null ? separator.toString() : 'null';
```

Comment:
you don't need to check separator on null value
try to delete it and check if your code will work in same way